agent/structs/structs.go

```diff
@@ -780,6 +791,37 @@ func (s *NodeService) IsSame(other *NodeService) bool {
 	return true
 }
 
+// IsSameService checks if one ServiceNode is the same as another, without looking
+// at the Raft information (that's why we didn't call it IsEqual). This is
+// useful for seeing if an update would be idempotent for all the functional
+// parts of the structure.
+// In a similar fashion as ToNodeService(), fields related to Node are ignored
+// see ServiceNode for more information.
+func (s *ServiceNode) IsSameService(other *ServiceNode) bool {
```

Comment:
Out of interest, why change the name from `IsSame` like we have in other places? Not a huge deal but I'm not sure I see the reason.

Reply:
Since I am not checking for Node information within (I saw comments in ToServiceNode() regarding this), I find this less confusing for a future user of the function. (I would expect IsSame() to behave like other -> taking into account all fields but RaftIndex.*

Reply:
Ah OK, maybe the doc comment could make that link a bit clearer since now it uses the same description of the purpose as the other IsSame methods:

```
// IsSameService checks if one ServiceNode is the same as another, without looking
// at the Node or Raft information (that's why we didn't call it IsEqual). This is
// useful for seeing if an update would be idempotent for all the service-related
// parts of the structure. Node fields are ignored since they are empty in the state store and populated on display. See ServiceNode for more information.
```